src/components/QuestionTimer.jsx

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from "react";
+
+export default function QuestionTimer({ timeout, onTimeout }) {
+    const [remainingTime, setRemainingTime] = useState(timeout)
+
+    useEffect(() => {
+        setTimeout(onTimeout, timeout)
+    }, [timeout, onTimeout])
+
+    useEffect(() => {
+        setInterval(() => {
+            setRemainingTime(prevRemainingTime => prevRemainingTime - 100)
+        }, 100)
+    }, [])
+
+    return(
+        <>
+            <progress id='question-time' max={timeout} value={remainingTime} />
+        </>
+    )
+
+}
```

Comment:
https://medium.com/@conboys111/why-you-should-always-clear-javascript-timers-settimeout-setinterval-7df1ec7ae880